core/src/services/gdrive/core.rs

```diff
@@ -46,15 +46,7 @@ pub struct GdriveCore {
     pub signer: Arc<Mutex<GdriveSigner>>,
 
     /// Cache the mapping from path to file id
-    ///
-    /// Google Drive uses file id to identify a file.
-    /// As the path is immutable, we can cache the mapping from path to file id.
-    ///
-    /// # Notes
-    ///
-    /// - The path is rooted at the root of the Google Drive.
-    /// - The path is absolute path, like `foo/bar`.
-    pub path_cache: Arc<Mutex<HashMap<String, String>>>,
+    pub path_cache: PathCacher<GdrivePathQuery>,
```

Comment:
Could this concrete struct, `PathCacher` be a trait instead?

Because the consumer of OpenDAL may want to implement a different cache for paths to support features like persisting paths into the disk.

Reply:
SGTM. We'd better not to rely on a specific implement. 🤔

Reply:
I plan to change into accepting an Operator instead. But I don't think it's a blocker of current PR. I won't introduce this change in this PR.

Reply:
I agree!